libs/components/pages/src/lib/modules/action-hub/types/action-hub-needs-attention.ts

```diff
@@ -16,11 +17,22 @@ export interface SkyActionHubNeedsAttention {
   /**
    * Specifies a link to resolve the action item.
    */
-  permalink: {
+  permalink?: {
     route?: {
       commands: any[];
       extras?: NavigationExtras;
     };
     url?: string;
   };
+  /**
+   * Specifies the modal parameters.
+   */
+  modal?: {
+    component: any;
+    config?: SkyModalConfigurationInterface;
+  };
+  /**
+   * Specifies a click handler for the action item.
+   */
+  click?: () => void;
```

Comment:
Should we pass some kind of context to this function so the consumer knows which item is clicked?

Reply:
I'm not sure what would be needed in the context that they don't already have. I'm inclined to keep the interface simple.